docs/concepts.md

```diff
@@ -1,27 +1,213 @@
-## Concepts
+# Concept Reference
 
-The ksonnet framework is centered around a couple of key concepts.
+The ksonnet framework is centered around a couple of key concepts and terms. This reference gives a brief explanation of each. To get a more tangible sense of how these concepts tie together, see the Tutorial (WIP, link pending).
+
+* Ksonnet
+  * [Application](#application)
+  * [Environment](#environment)
+  * [Component](#component)
+  * [Prototype](#prototype)
+  * [Parameter](#parameter)
+  * [Part](#part)
+  * [Package](#package)
+  * [Registry](#registry)
+* Related
+  * [Manifest](#manifest)
+  * [Jsonnet](#jsonnet)
+
+
+The following diagram shows how the ksonnet framework works holistically:
+
+![ksonnet overview diagram](/docs/img/ksonnet_overview.svg)
```

Comment:
Would it be possible to parallelize the horizontal arrows (---->) and make them the same width/distance? Initial look made me think that `component` was a part of  the first grouping.

I think it would also be helpful if all the components were the same shape as `component` (the cylinder).